spec/models/mixins/inter_region_api_method_relay_spec.rb

```diff
@@ -237,29 +255,68 @@ def expect_api_call(expected_action, expected_args = nil)
           expect(api_collection).to receive(action) do |args, &block|
             expect(args).to eq(expected_args)
             expect(block.call).to eq("some stuff")
-          end
+          end.and_return(api_success_result)
+
           described_class.exec_api_call(region, collection_name, action, expected_args, &resource_proc)
         end
 
         it "defaults the args to an empty hash" do
           expect(api_collection).to receive(action) do |args, &block|
             expect(args).to eq({})
             expect(block.call).to eq("some stuff")
-          end
+          end.and_return(api_success_result)
+
           described_class.exec_api_call(region, collection_name, action, &resource_proc)
         end
 
         it "defaults the args to an empty hash when nil is explicitly passed as args" do
           expect(api_collection).to receive(action) do |args, &block|
             expect(args).to eq({})
             expect(block.call).to eq("some stuff")
-          end
+          end.and_return(api_success_result)
+
           described_class.exec_api_call(region, collection_name, action, nil, &resource_proc)
         end
       end
     end
   end
 
+  describe ".instance_for_resource" do
+    let(:collection) { double("ManageIQ::API::Client::Collection", :name => collection_name) }
+    let(:resource)   { double("ManageIQ::API::Client::Resource", :collection => collection, :id => 10) }
```

Comment:
Since we run in region 1, can you create a record in region 1?  Or something like `MiqRegion.id_in_region(10, MiqRegion.my_region_number)`